Builder zero-output failures should surface post-mortem in validation error message

## Problem

When the builder CLI starts but produces zero output (e.g., due to rate limiting or usage caps), the validation error message only reports worktree state but not the CLI-level diagnosis:

```
[ERROR] builder phase validation failed: worktree exists (branch=feature/issue-2775, 
commits_ahead=0, uncommitted=none); remote branch missing; no PR; 
labels=[loom:curated, loom:building]; checkpoint=planning; 
log=/Users/.../loom-builder-issue-2775.log
```

The post-mortem diagnostics system (#2780) collects structured zero-output analysis (exit codes, CLI lifetime, rate limit detection, sentinel status) and stores it in `ctx.last_postmortem`. The `_gather_diagnostics` method already includes it as `diag["postmortem"]`. But this data is **not surfaced** in the diagnostic summary string or error message, so the failure is classified as `builder_unknown_failure` instead of something actionable.

## Expected Behavior

The error message should include the post-mortem analysis when available:

```
[ERROR] builder phase validation failed: worktree exists (...); checkpoint=planning; 
postmortem: CLI exited in 5s with 0 chars output, exit_code=0, no rate_limit_sentinel; 
log=...
```

This would make `builder_unknown_failure` much more actionable — operators could immediately distinguish:
- Rate limit / usage cap (CLI started, exited fast, no output)
- Auth failure (CLI started, auth check failed)
- MCP failure (CLI started, MCP init failed)
- Legitimate zero-changes (CLI ran long enough to analyze)

## Implementation

1. In `_gather_diagnostics` or `_build_diagnostic_summary`, include `diag["postmortem"]` data in the summary string
2. Consider using the post-mortem cause to refine the systematic failure pattern from `builder_unknown_failure` to something more specific (e.g., `builder_rate_limited`, `builder_cli_crash`)

## Relevant Files

- `loom-tools/src/loom_tools/shepherd/phases/builder.py` — `_gather_diagnostics()` (line ~3230), `_build_diagnostic_summary()`, post-mortem handling (line ~3323)
- `loom-tools/src/loom_tools/shepherd/phases/base.py` — `run_worker_phase()` post-mortem collection

## Context

From shepherd run for #2775 on 2026-02-18. The builder log showed `# CLAUDE_CLI_START` and nothing else — the CLI likely hit a rate/usage limit and exited immediately. Three consecutive failures were all the same root cause but classified generically.

---
🤖 Generated with [Claude Code](https://claude.com/claude-code)

--- a/loom-tools/src/loom_tools/shepherd/phases/builder.py
+++ b/loom-tools/src/loom_tools/shepherd/phases/builder.py
@@ -3613,6 +3613,11 @@ class BuilderPhase:
             parts.append(last_error)
         if diag["low_output_cause"]:
             parts.append(f"low_output_cause={diag['low_output_cause']}")
+        # Surface post-mortem analysis in the summary so operators can
+        # distinguish rate limits, auth failures, and MCP crashes from
+        # the validation error message alone.  See issue #2801.
+        if diag.get("postmortem") and diag["postmortem"].get("summary"):
+            parts.append(f"postmortem: {diag['postmortem']['summary']}")
         if diag["worktree_exists"]:
             if diag["has_uncommitted_changes"]:
                 uncommitted_note = f"{diag['uncommitted_file_count']} files"

--- a/loom-tools/tests/shepherd/test_phases.py
+++ b/loom-tools/tests/shepherd/test_phases.py
@@ -1183,6 +1183,7 @@ class TestBuilderDiagnostics:
         mock_context.config = ShepherdConfig(issue=42)
         mock_context.repo_root = tmp_path
         mock_context.last_low_output_cause = None
+        mock_context.last_postmortem = None
 
         builder = BuilderPhase()
 
@@ -1205,7 +1206,7 @@ class TestBuilderDiagnostics:
             diag = builder._gather_diagnostics(mock_context)
 
         summary = diag["summary"]
-        # Should have 5 semicolon-separated sections
+        # Should have 5 semicolon-separated sections (no postmortem when last_postmortem is None)
         parts = summary.split("; ")
         assert len(parts) == 5
         assert "worktree" in parts[0]
@@ -1214,6 +1215,45 @@ class TestBuilderDiagnostics:
         assert "labels=" in parts[3]
         assert "log=" in parts[4]
         assert diag["low_output_cause"] is None
+
+    def test_diagnostics_summary_includes_postmortem(
+        self, mock_context: MagicMock, tmp_path: Path
+    ) -> None:
+        """Postmortem summary should appear in diagnostic summary when present."""
+        mock_context.worktree_path = tmp_path / "nonexistent"
+        mock_context.config = ShepherdConfig(issue=42)
+        mock_context.repo_root = tmp_path
+        mock_context.last_low_output_cause = None
+        mock_context.last_postmortem = {
+            "summary": "CLI started but produced zero output; log duration: 5s; exit(wait=0)",
+            "cli_started": True,
+            "has_rate_limit": False,
+        }
+
+        builder = BuilderPhase()
+
+        def fake_run(cmd, **kwargs):
+            cmd_str = " ".join(str(c) for c in cmd)
+            if "gh" in cmd_str:
+                return subprocess.CompletedProcess(
+                    args=cmd,
+                    returncode=0,
+                    stdout="loom:building, loom:curated",
+                    stderr="",
+                )
+            return subprocess.CompletedProcess(
+                args=cmd, returncode=1, stdout="", stderr=""
+            )
+
+        with patch(
+            "loom_tools.shepherd.phases.builder.subprocess.run", side_effect=fake_run
+        ):
+            diag = builder._gather_diagnostics(mock_context)
+
+        summary = diag["summary"]
+        assert "postmortem:" in summary
+        assert "CLI started but produced zero output" in summary
+        assert "log duration: 5s" in summary
 
 
 class TestBuilderQualityValidation:

